src/app/view/components/dot-toolbar/dot-toolbar.component.ts

```diff
@@ -33,7 +35,9 @@ export class ToolbarComponent implements OnInit {
 
     siteChange(site: Site): void {
         this.siteService.switchSite(site);
-        this.dotNavigationService.goToFirstPortlet();
+        if (this.route.snapshot['_routerState'].url.indexOf('edit-page/') >= 0) {
+            this.dotNavigationService.goToFirstPortlet();
```

Comment:
What we talked is that it should be to `site-browser` and if anything happen, lilke the user doesn't have access to the site browser the router will redirect to first portlet automatically.